server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java

```diff
@@ -75,17 +78,116 @@
+                                    List<PipelineAggregator> pipelineAggregators,
+                                    Map<String, Object> metaData) throws IOException {
+
+                ExecutionMode execution = null;
+                if (executionHint != null) {
+                    execution = ExecutionMode.fromString(executionHint, deprecationLogger);
+                }
+                if (valuesSource instanceof ValuesSource.Bytes.WithOrdinals == false) {
+                    execution = ExecutionMode.MAP;
+                }
+                if (execution == null) {
+                    execution = ExecutionMode.GLOBAL_ORDINALS;
+                }
+
+                if ((includeExclude != null) && (includeExclude.isRegexBased()) && format != DocValueFormat.RAW) {
+                    throw new IllegalArgumentException("Aggregation [" + name + "] cannot support regular expression style "
+                        + "include/exclude settings as they can only be applied to string fields. Use an array of values for "
+                        + "include/exclude clauses");
+                }
+
+                return execution.create(name, factories, valuesSource, format, bucketCountThresholds, includeExclude, context, parent,
+                    significanceHeuristic, sigTermsFactory, pipelineAggregators, metaData);
+
+            }
+        };
+    }
+
+    /**
+     * This supplier is used for all fields that expect to be aggregated as a numeric value.
+     * This includes floating points, and formatted types that use numerics internally for storage (date, boolean, etc)
+     */
+    private static SignificantTermsAggregatorSupplier numericSupplier() {
+        return new SignificantTermsAggregatorSupplier() {
+            @Override
+            public Aggregator build(String name,
+                                    AggregatorFactories factories,
+                                    ValuesSource valuesSource,
+                                    DocValueFormat format,
+                                    TermsAggregator.BucketCountThresholds bucketCountThresholds,
+                                    IncludeExclude includeExclude,
+                                    String executionHint,
+                                    SearchContext context,
+                                    Aggregator parent,
+                                    SignificanceHeuristic significanceHeuristic,
+                                    SignificantTermsAggregatorFactory sigTermsFactory,
+                                    List<PipelineAggregator> pipelineAggregators,
+                                    Map<String, Object> metaData) throws IOException {
+
+                if ((includeExclude != null) && (includeExclude.isRegexBased())) {
+                    throw new AggregationExecutionException("Aggregation [" + name + "] cannot support regular expression style include/exclude "
```

Comment:
Just noticed I forgot to change this, should be IllegalArgException, will fix in the morning :)